MAINTAINERS: add initial entries for west.yml projects

mbolivar-nordic opened the pull request:
This is part of the road towards replacing CODEOWNERS with
MAINTAINERS (tracked in the process working group as #38725).

As part of that work, we wanted to have a way to track the maintainers
of each west project (entry in west.yml) by name of project.

This is an initial attempt at that, based mostly on my personal
knowledge, the git logs, and some rough guesswork. I fully expect
omissions and errors, but it should be enough to get us started, with
fixes to follow incrementally from the people who know their
individual areas better than I do.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

stephanosio reviewed (commented):
It bothers me a little that this file uses 4 spaces for indentation (which, I understand, is very questionable) yet the changes here use 2 spaces.

I guess we can update the whole file to use 2 spaces instead of 4 spaces later.

carlescufi commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+    - manifest-tinycbor
+
+"West project: tinycrypt":
+  status: obsolete
```
Hmm no, this is not obsolete. There are no viable plans or alternatives to replace it (yet).

mbolivar-nordic replied:
> Hmm no, this is not obsolete. There are no viable plans or alternatives to replace it (yet).

Yeah -- I wrote this patch before I wrote https://github.com/zephyrproject-rtos/zephyr/commit/89b07084d668fbc14da5ae1dbc7302b674c5633a. Without that patch, the only thing that fit areas without even a single collaborator was "obsolete". I will update these now that that patch is merged and we are OK with having "odd fixes" be a status for things without collaborators.

carlescufi commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+    - manifest-cmsis
+
+"West project: edtt":
+  status: odd fixes
```
I wonder if @aescolar would volunteer to maintain this one?

carlescufi commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+    - manifest-fatfs
+
+"West project: hal_altera":
+  status: obsolete
```
Checking with @nashif about this one

nashif replied:
how was this determined? it is not obsolete.

mbolivar-nordic replied:
@nashif who is the collaborator then? Does it have any?

nashif replied:
you can put me.

mbolivar-nordic replied:
OK. https://github.com/zephyrproject-rtos/zephyr/pull/44614#discussion_r846299080 for details on how it was determined originally

carlescufi commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+    - manifest-hal_atmel
+
+"West project: hal_cypress":
+  status: odd fixes
```
pinging @npal-cy, should you be the maintainer?

npal-cy replied:
it should be @ifyall for both: hal_cypress/hal_infineon

carlescufi commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+    - manifest-hal_gigadevice
+
+"West project: hal_infineon":
+  status: odd fixes
```
Same about this one @npal-cy

npal-cy replied:
it should be @ifyall for both: hal_cypress/hal_infineon

carlescufi commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+  status: odd fixes
+  collaborators:
+    - cfriedt
+    - gmarull
```
@vaishnavachath would you like to be added here?

carlescufi commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+
+"West project: lvgl":
+  status: odd fixes
+  files:
```
@brgl would you like to maintain or be added as a collaborator?

brgl replied:
Sure, would be great. I would also love to be able to add labels to PRs and launch CI workflows for first-time contributors in the main zephyr tree if possible.

jfischer-no replied:
Are display collaborators not asked in this regard?

mbolivar-nordic replied:
> Are display collaborators not asked in this regard?

@jfischer-no If you are volunteering for the role, please say so clearly and I will add you

mbolivar-nordic replied:
> Sure, would be great. I would also love to be able to add labels to PRs and launch CI workflows for first-time contributors in the main zephyr tree if possible.

@brgl awesome! Please read https://docs.zephyrproject.org/latest/project/project_roles.html#contributor and nominate yourself for Triage role as Contributor or as Collaborator as you see fit.

carlescufi commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+  maintainers:
+    - d3zd3z
+  collaborators:
+    - nvlsianpu
```
Andrzej should be a maintainer

carlescufi commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+
+"West project: segger":
+  status: odd fixes
+  files: []
```
Can you add @nordic-krch as a collaborator here?

carlescufi commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+  maintainers:
+    - microbuilder
+  collaborators:
+    - joerchan
```
Please add @SebastianBoe as well

microbuilder replied:
@theotherjimmy as well

carlescufi commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+  maintainers:
+    - microbuilder
+  collaborators:
+    - joerchan
```
Please add @SebastianBoe as well

microbuilder replied:
@theotherjimmy as well

carlescufi commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+  maintainers:
+    - microbuilder
+  collaborators:
+    - joerchan
```
Please add @SebastianBoe as well

microbuilder replied:
@theotherjimmy as well

carlescufi reviewed (dismissed):
Thank you!

nashif commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+"West project: cmsis":
+  status: odd fixes
+  collaborators:
+    - nashif
```
not me

brgl commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+    - manifest-loramac-node
+
+"West project: lvgl":
+  status: odd fixes
```
If you want me as maintainer then you can change the status to `maintained`

stephanosio replied:
maybe @microbuilder (ARM maintainer) and @povergoing (from ARM)

stephanosio commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+    - "area: Networking"
+    - "area: civetweb"
+
+"West project: cmsis":
+  status: odd fixes
+  collaborators:
+    - nashif
```
maybe @microbuilder (ARM maintainer) and @povergoing (from ARM) as collaborators

povergoing replied:
OK for me

keith-zephyr commented on MAINTAINERS.yml:
```diff
@@ -1841,6 +1838,489 @@ West:
+    - MulinChao
+    - ChiHuaL
+    - WealianLiao
+    - sjg20
+    - keith-zephyr
+    - jackrosenthal
+    - fabiobaltieri
+    - yperess
```
The hal_nuvoton module is for older EC chips not supported by Chromebooks.  The newer chips are in the tree. So it's probably better to only include the Nuvoton folks and you can remove these individuals
```
    - sjg20
    - keith-zephyr
    - jackrosenthal
    - fabiobaltieri
    - yperess
    ```

mbolivar-nordic committed:
MAINTAINERS: add initial entries for west.yml projects

This is part of the road towards replacing CODEOWNERS with
MAINTAINERS (tracked in the process working group as #38725).

As part of that work, we wanted to have a way to track the maintainers
of each west project (entry in west.yml) by name of project.

This is an initial attempt at that, based mostly on my personal
knowledge, the git logs, and some rough guesswork. I fully expect
omissions and errors, but it should be enough to get us started, with
fixes to follow incrementally from the people who know their
individual areas better than I do.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic commented:
Thanks for all the reviews. Comments should be addressed, please revisit.

mbolivar-nordic commented:
Rebased to address merge conflict.
